pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/forward/ForwardIndexReaderFactory.java

```diff
@@ -106,6 +108,10 @@ public static ForwardIndexReader createRawIndexReader(PinotDataBuffer dataBuffer
           : new FixedByteChunkSVForwardIndexReader(dataBuffer, storedType);
     }
 
+    if (version >= VarByteChunkForwardIndexWriterV5.VERSION) {
```

Comment:
This should be exact match. Let's make it a switch

Reply:
Done